operator/src/main/java/org/keycloak/operator/v2alpha1/NameUtils.java

```diff
@@ -0,0 +1,46 @@
+package org.keycloak.operator.v2alpha1;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+public class NameUtils {
+
+    // DNS Subdomain Names
+    // https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
+    // TODO: test me
+    public static String getValidDNSSubdomainName(String in) {
```

Comment:
cc. @metacosm @csviri
would you want to integrate those kind of generic utilities in the SDK itself?

Reply:
There's one such utility in the fabric8 client, iirc. `HasMetadata` has a regexp that checks finalizers (which include a DNS subdomain name as the first part) also.

Reply:
Cannot find it, BTW this is "automatic-mangling" of names (and probably we would need something similar in the CRD generator too ...)

Reply:
Thanks for the reference!